### ISSUE TITLE
Doesn't build on clang

### ISSUE THREAD

**Issue by KenanSulayman** [centaurean/sharc#24]
Hey,

I just tried to deploy sharc on a FreeBSD installation and couldn't get it to compile w/o using gcc5. The actual error is strange:

```
Linking Sharc ...
*.o -> sharc
./src/header.o: file not recognized: File format not recognized
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:76: recipe for target 'sharc' failed
```

...

```
user@host# file src/header.o 
src/header.o: LLVM IR bitcode
```

welp :smirk: 

It works, however, when installing *gcc5* (either per port or pkg) and then doing a `CC=gcc5 CXX=g++5 gmake all` (note *gmake*, since *BSD make* went bonkers).

:v:

**Comment by gpnuma** [comment]
Hello,
It looks like the clang compiler on FreeBSD produces files which are not usable by the default linker. That could potentially come from link time optimization switches such as -flto, removing them might fix this problem.
Another option would be to try to use the linker provided by the clang utilities - although it might just be a command frontend to the system linker - and see if it improves things.

**Comment by KenanSulayman** [comment]
Nope, doesn't work and I actually didn't have the time to investigate this behaviour thoroughly. It didn't fix it and AFAIK LLVM falls back to the gcc linker anyways. I gave it up and exploit the linux ABI compatibility of FreeBSD by running the binaries built on Ubuntu 15.04 - I'd love to use a native binary, though.
